### DISCUSSION TITLE
ClusterCompare vs. ClusterBLAST

### DISCUSSION THREAD

**Discussion by nicholascdove** [antismash/antismash#655] [General]
Thanks for the great program! I had a question on the two BGC comparison tools. My understanding is that ClusterCompare is now the preferred algorithm for making comparisons. However, when looking at erythromycin (BGC0000054), Known Cluster Blast gives a 64% similarity to erythromycin A (BGC0000055) , but erythromycin A is not represented at all in the ClusterCompare MI-BiG output. While the top ClusterCompare hit has a lowish similarity (oronofacic acid - Protocluster to region: 0.41), the 2D structure of the compound doesn't look very similar to erythromycin. Also, while erythromycin and erythromycinA are slightly different, I would expect ClusterCompare to rank them as similar. 

Is my understanding flawed? Should we be using both ClusterCompare and ClusterBLAST to understand BGC similarity?

Here's the AntiSMASH record for BGC0000054: https://mibig.secondarymetabolites.org/repository/BGC0000054/generated/index.html#r1c1

Thanks!

**Comment by SJShaw** [comment]
When the accession of the query matches the accession of the reference, it's excluded because the results will always be a perfect match. So if you use the genbank provided by MIBiG for BGC0000055 as an input to antiSMASH without editing the accession, it will never show BGC0000055 as a match.

In terms of comparing the two clusters, there's quite a lot going on. There's missing genes, inserted genes, poor identity matching, and different ordering of genes.

For context, this is what the comparison of the two looks like, with BGC0000055 as the query:
![image](https://github.com/antismash/antismash/assets/1700735/7677565c-dd36-4a4c-b3a9-83888eb52fbf)

I think there's some extra score penalties that could be applied when only a single gene in the reference hits, say halving the ordering score (at least when the reference contains multiple genes), which would move 54 to 5th place instead of 15th. However, that would still leave entries such as oronofacic acid higher in the ranking for the "reference in query" mode of ClusterCompare (the default) as it's fully contained within the query.

In the end, a combination of the two is useful at the moment, but a human expert will always be required, as it only takes a couple of amino acids to change in the gene translations to have a completely different product for some of these modular cluster types.

**Comment by nicholascdove** [comment]
Thanks very much for this insightful and thorough answer! This explanation has really helped my understanding. Thanks!
